### DISCUSSION TITLE
postUpgradeTask using functions

### DISCUSSION THREAD

**Discussion by mjnagel** [renovatebot/renovate#14287] [Request Help]
I'm trying to run a script as a post upgrade task. For purposes of an easy question:
- `foo` is a function defined in `/tmp/functions.sh` which runs a number of commands - all of which are built in commands (i.e. things like cat, sed, grep, and awk)
- `config.js` has `allowedPostUpgradeCommands` set to `foo`
- `renovate.json` for the repo has `foo` as its post upgrade command

I essentially am just looking for how I make this setup work. I'm using the docker image to run renovate, and have validated that `/tmp/functions.sh` and `config.js` look as expected inside the container. I've tried the following with no luck:
- `source /tmp/functions.sh` in the shell before running `renovate`
- adding `source /tmp/functions.sh` to `~/.profile` and `~/.bashrc`, setting ENV=~/.profile
- 
Regardless I always get `/bin/sh: foo: command not found`

It seems like renovate is running a bourne shell for these, but I'm not sure how I would go about getting things sourced in it. The only thing that does seem to work is setting `source /tmp/functions.sh && foo` as the postUpgradeTask (and allowedPost...). That's not extremely clean but totally understand if that is what is needed.

**Comment by viceice** [comment]
Not possible to call a shell function from renovate. Create a wrapper shell script in path and call your function from there.
